File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] ret = new int[2];
        Map<String, Integer> mp = new HashMap<>();
        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다. 
        // System.out.println("Hello Java");
        String prev = "" + words[0].charAt(0);
        for(int i = 0; i < words.length; i++){    
            // 앞의 첫글자랑 다르면
            if(prev.charAt(prev.length() - 1) != words[i].charAt(0)){
                ret[0] = (i % n) + 1; //몇번째인지
                ret[1] = (i / n) + 1; //회차
                break;
            }
            //나온적있는 단어
            if(mp.containsKey(words[i])){
                ret[0] = (i % n) + 1;
                ret[1] = (i / n) + 1;
                break;
            }
            // 단어의 길이가 1이면
            if(words[i].length() == 1){
                ret[0] = (i % n) + 1;
                ret[1] = (i / n) + 1;
                break;
            }
            prev = words[i];
            mp.put(words[i], 1);
        }
        
        return ret;
    }
}
// 조건1: 앞사람의 마지막 단어가 시작인지? -> prev로 확인
// 조건2: 이전 등장했던 단어인가?
// 조건3: 한 글자인 단어는 인정x
// {첫탈락사람번호, 몇번째}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    st = set()
    prev = words[0][0]
    for i,x in enumerate(words):
        if prev[-1] != x[0] or x in st:
            return [i%n+1, i//n+1]
        st.add(x)
        prev = x
    return [0, 0]
    
"""
[조건]
1~n, 
1->2-> ...  차례대로 
마지막 다음 1, 마지막 문자로 시작, 이전 등장단어 사용불가, 한글자x

가장먼저 탈락하는 사람 번호, 차례
사람수, 2 <= n <=10

[아이디어]
순회하면서 조건을 어기면 해당 인덱스의 
번호 = %한후 +1
차례 = / + 1


[자료구조]

[로직]


"""